### PR TITLE
Support 'dom' Platform.OS for react-native-dom

### PR DESCRIPTION
**Pull request by peacechen**
## Description

This PR adds support for `react-native-dom`, a more comprehensive library to run RN apps in the browser.  The change is a simple one-liner addition to check whether `Platform.OS === 'dom'`.

@machour 
It looks like there's a lot of cruft in this PR, but it's because my original PR was never merged.  I've sync'd with master and merged in order to submit another PR.

## Compatibility

Web / DOM

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
rest N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,6 +186,12 @@ include ':app'
 
 </details>
 
+<details>
```
Comment (machour) [inline README.md]: I don't think this is needed in `Linking`, as I already added a section at the bottom to the README mentioning this.

Could you possibly take a look at https://github.com/rebeccahughes/react-native-device-info#react-native-web and mention "react-native-dom" in it? (feel free to rename the section title)

**Comment by machour** [comment]
PS: And thank you for handling this so quickly after react-native-dom release 🙌

File: README.md (hunk 1 above)
Comment (peacechen) [reply README.md]: That section must've gotten pulled in from the merge.  I'll remove it and add rn-dom to the bottom.

**Comment by peacechen** [comment]
Update the Readme.  Recommend squashing this PR during merge so that all the intermediary merges don't uglify your commit tree :)

**Comment by mikehardy** [comment]
@machour this seems good to go with a need for a changelog entry in the "next" header I have building up. Any reason not to merge? If I don't hear dissent I'll merge it in a day or two

**Comment by machour** [comment]
This PR just fell into the crack of my mind. Feel free to merge it @mikehardy 
Sorry for the time wasted @peacechen!

**Comment by mikehardy** [comment]
It's out!

https://www.npmjs.com/package/react-native-device-info/v/1.2.0

**Pull request merged**